tests/registry/download.py

```diff
@@ -1,11 +1,18 @@
 import io
```

Comment:
I may have missed it, but should there be a test for the glob stuff?

Reply:
there is a download test the uses the code, but I aim to refactor it and test it more heavily then, as it currently is it's so io bound that it's a pain to write a meaningful test (which is why I need to refactor it)